app/models/components/course/duplicator_component_host.rb

```diff
@@ -0,0 +1,15 @@
+class Duplicator::ModelComponentHost
```

Comment:
It is camel case, but your filename does not match.

If you say it's strange for it to be part of the course model host, then maybe you should mix in behaviour to the course model host instead?